text/trace/0220-messaging-semantic-conventions-span-structure.md

```diff
@@ -0,0 +1,657 @@
+called by messaging SDKs to forward messages to the application.
+
+A "Deliver" span covers the call of such a callback or handler and should link
+to the creation context of all messages that are forwarded via the respective
+call.
+
+#### Instrumenting pull-based scenarios
+
+In pull-based consumer scenarios, the delivery of messages is requested by the
+application code. This usually involves a blocking call, which returns zero or
+more messages on completion.
+
+A "Receive" span covers such calls and should link to the creation context of
+all messages that are forwarded via the respective call.
+
+#### General considerations for both push-based and pull-based scenarios
+
+The operations modelled by "Deliver" or "Receive" spans do not strictly refer
+to receiving the message from intermediaries, but instead refer to the
+application receiving messages for processing. If messages are fetched from the
+intermediary and forwarded to the application in one go, the whole operation
+might be covered by a "Deliver" or "Receive" span. However, libraries or SDKs
+might pre-fetch messages from intermediaries and cache those messages, and only
+forward messages to the application at a later time. In this case, the
+operation of pre-fetching and caching should not be covered by the "Deliver" or
+"Receive" spans.
+
+Operations covered by "Deliver" or "Receive" can forward zero messages (e. g.
+to notify the application that no message is available for processing), one
+message, or multiple messages (a batch of messages). "Deliver" and "Receive"
+spans should link to the creation context of the messages forwarded, thus those
+spans can link to zero, one, or multiple producer spans.
+
+For single-message scenarios, and if the "Deliver" or "Receive" spans would be
+root spans of a new trace, the creation context may also be used as a parent on
+those operations in addition to being added as a link. Keeping single-messages
+operations in the same trace can greatly improve the user experience.
+
+#### Settlement
+
+Messages can be settled in a variety of different ways:
+
+* The intermediary settles the messages as it is sent to the consumer. No
+  settlement operations happen on the consumer.
+* The consumer settles a message without awaiting an acknowledgment from the
+  intermediary.
+* The consumer settles a message and awaits an acknowledgment from the
+  intermediary. This involves a round-trip exchange between the consumer and
+  the intermediary.
+
+Settlement operations on the consumer can either be triggered manually by the
+user, or can be triggered automatically by messaging SDKs based on return
+values of callbacks.
+
+A "Settle" span should be created for every settlement operation that happens
+on the consumer (at-least-once and exactly-once). SDKs will, in some cases,
+auto-settle messages in push-scenarios when messages are delivered via
+callbacks. In cases where it is possible, it is recommended to create the
```

Comment:
```suggestion
callbacks. 
```

I suggest removing this sentence as it's misleading. If settlement happens in scope of deliver span - great, deliver will be parent automatically. Otherwise, it might be confusing since deliver can end before settlement happens or they may happen in different scopes. Instrumentation authors should not try to fix it. 
I believe we already have an example that demonstrates that it can happen on its own and it should be good enough.